workflow/validate/validate.go

```diff
@@ -245,6 +250,16 @@ func resolveAllVariables(scope map[string]interface{}, tmplStr string) error {
 	return unresolvedErr
 }
 
+// checkValidWorkflowVariablePrefix is a helper methood check variable starts workflow root elements
+func checkValidWorkflowVariablePrefix(tag string) bool {
+	for _, rootTag := range common.GlobalVarValidWorkflowVariablePrefix {
+		if strings.HasPrefix(tag, rootTag) {
```

Comment:
Can we also add `.` to HasPrefix (e.g. rootTag + '.') so that things like `stepsfoo` won't match but `steps.` will match?

Reply:
good catch. Thanks. I will fix it